src/components/stories/Stories.jsx

```diff
@@ -1,20 +1,141 @@
 import userData from "./UserData";
-import { Link } from "react-router-dom";
-import { Row, Button } from "react-bootstrap";
+import { useState } from "react";
+import { Form, Row, Button, Modal } from "react-bootstrap";
 import UserCard from "../layout/UserCard";
+import { useInsert } from "react-supabase";
+import { useAuth } from "../../context/SupaContext";
+import { FetchingButton } from "../layout/Buttons/FetchingButton";
+import { CountryDropdown, RegionDropdown } from "react-country-region-selector";
 import "./Stories.css";
 export default function Stories() {
+  const [show, setShow] = useState(false);
+  const [country, setCountry] = useState("");
+  const [region, setRegion] = useState("");
+  const [title, setTitle] = useState("");
+  const [desc, setDesc] = useState("");
+  const [err, setErr] = useState("");
+  const { user } = useAuth();
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+  const [{ fetching }, execute] = useInsert("stories");
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const { error } = await execute({
+      user_id: user.id,
+      title: title,
+      description: desc,
+      region: region,
+      country: country,
+    });
+    if (!error) {
+      // insert successfully
+      clearInput();
+      handleClose();
+    } else {
+      // display error
+      setErr(error.message);
+    }
+  };
+
+  const clearInput = () => {
+    setTitle("");
+    setDesc("");
+    setCountry("");
+    setRegion("");
+    setErr("");
+  };
+
   return (
     <div className="stories">
       <div className="main_title">
         <h1>User Story</h1>
-        <Button className="story_btn" as={Link} to={"/404/"}>
+        <Button className="story_btn" onClick={handleShow}>
           Create your Story
         </Button>
```

Comment:
Show this button to only logged in users.

Show a button with a link to the login page so users can login